src/Pact/Types/Term.hs

```diff
@@ -285,46 +320,38 @@ data NativeDFun = NativeDFun
   }
 
 instance Eq NativeDFun where a == b = _nativeName a == _nativeName b
-instance Show NativeDFun where show a = show $ _nativeName a
+instance Show NativeDFun where
+  showsPrec p (NativeDFun name _) = showParen (p > 10) $
```

Comment:
why showsPrec/showParen here? It's just showing the name.

Reply:
As written it renders as `NativeDFun name _`. Say the context is `Just _`, so the surrounding precedence would be 11 and it renders as `Just NativeDFun name _`

Reply:
Sorry, I don't understand "precedence" here ... what is up with the magic number 10?

Reply:
The [`Show` typeclass](http://hackage.haskell.org/package/base-4.12.0.0/docs/Text-Show.html#t:Show) has a good explanation